hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/replication/TestContainerImporter.java

```diff
@@ -187,6 +193,36 @@ public void testInconsistentChecksumContainerShouldThrowError() throws Exception
         contains("Container checksum error"));
   }
 
+  @Test
+  public void testImportContainerScansContainer() throws Exception {
+    long containerId = 1;
+    try (MockedStatic<OnDemandContainerDataScanner> mockedStatic = mockStatic(OnDemandContainerDataScanner.class)) {
+      // create container
+      KeyValueContainerData containerData = new KeyValueContainerData(containerId,
+          ContainerLayoutVersion.FILE_PER_BLOCK, 100, "test", "test");
+      KeyValueContainer container = new KeyValueContainer(containerData, conf);
+      ContainerController controllerMock = mock(ContainerController.class);
+      when(controllerMock.importContainer(any(), any(), any())).thenReturn(container);
+
+      // create containerImporter object
+      ContainerSet containerSet = newContainerSet(0);
+      MutableVolumeSet volumeSet = new MutableVolumeSet("test", conf, null,
+          StorageVolume.VolumeType.DATA_VOLUME, null);
+      HddsVolume targetVolume = mock(HddsVolume.class);
+      doNothing().when(targetVolume).incrementUsedSpace(anyLong());
+      ContainerImporter containerImporter = new ContainerImporter(conf,
+          containerSet, controllerMock, volumeSet, volumeChoosingPolicy);
+
+      // import the container
+      File tarFile = containerTarFile(containerId, containerData);
+      containerImporter.importContainer(containerId, tarFile.toPath(),
+          targetVolume, NO_COMPRESSION);
+
+      // verify static method was called
+      mockedStatic.verify(() -> OnDemandContainerDataScanner.scanContainer(container), times(1));
```

Comment:
We should also verify in the test that the container actually gets added to the containerSet.
Something like,
```suggestion
      mockedStatic.verify(() -> OnDemandContainerDataScanner.scanContainer(container), times(1));
      assertNotNull(containerSet.getContainer(containerId));
```